compiler/optimizer/Inliner.cpp

```diff
@@ -2617,7 +2617,8 @@ TR_TransformInlinedFunction::transform()
    // If the first block has exception predecessors or multiply predecessors then we can't merge
    // the first block with the caller's block
    //
-   if ((firstBlock->getPredecessors().size() > 1) ||
+   if (comp()->isJProfilingCompilation() ||
+       (firstBlock->getPredecessors().size() > 1) ||
```

Comment:
Could you please add a brief comment here on the reason why this logic needs to be guarded by the extra test you added ?